sklearn/ensemble/_forest.py

```diff
@@ -679,6 +681,36 @@ def feature_importances_(self):
         all_importances = np.mean(all_importances, axis=0, dtype=np.float64)
         return all_importances / np.sum(all_importances)
 
+    def _get_estimators_indices(self):
+        # Get drawn indices along both sample and feature axes
+        for tree in self.estimators_:
+            if not self.bootstrap:
+                yield np.arange(self._n_samples, dtype=np.int32)
+            else:
+                seed = tree.random_state
+                # Operations accessing random_state must be performed identically
+                # to those in `_parallel_build_trees()`
+                sample_indices = _generate_sample_indices(
+                    seed, self._n_samples, self._n_samples_bootstrap
+                )
+
+                yield sample_indices
+
+    @property
+    def estimators_samples_(self):
+        """
+        The subset of drawn samples for each base estimator.
+
+        Returns a dynamically generated list of indices identifying
+        the samples used for fitting each member of the ensemble, i.e.,
+        the in-bag samples.
+
+        Note: the list is re-created at each call to the property in order
+        to reduce the object memory footprint by not storing the sampling
+        data. Thus fetching the property may be slower than expected.
+        """
+        return [sample_indices for sample_indices in self._get_estimators_indices()]
```

Comment:
I would rather return a 2D array rather than a list of arrays. This means that we could remove the `_get_estimators_indices` generator since it does not bring any memory usage benefit. This property could allocate the result array ahead of time with `np.empty` and then fill it column by column with a simple for loop without yield expressions.

Reply:
Same comment here: `Bagging*` models already use `estimators_samples_` as a list of arrays. So maybe we should stay consistent.

Reply:
Another issue is that `sample_indices` may not be consistent across estimators. So this might result in a ragged 2D array.